tests/org.palladiosimulator.dataflow.confidentiality.analysis.tests/src/org/palladiosimulator/dataflow/confidentiality/analysis/DataFlowConfidentialityAnalysisTest.java

```diff
@@ -2,58 +2,99 @@
+	 */
+	@BeforeAll
+	public static void initializeAnalysis() {
         final var usageModelPath = Paths.get("models", "BranchingOnlineShop", "default.usagemodel").toString();
         final var allocationPath = Paths.get("models", "BranchingOnlineShop", "default.allocation").toString();
-
-        final DataFlowConfidentialityAnalysis analysis = new StandalonePCMDataFlowConfidentialtyAnalysis(
+        
+        analysis = new StandalonePCMDataFlowConfidentialtyAnalysis(
                 TEST_MODEL_PROJECT_NAME, Activator.class, usageModelPath, allocationPath);
 
         analysis.initalizeAnalysis();
+	}
+
+	/**
+	 * Tests the standalone analysis by executing the ActionSequenceFinder. Fails if no sequences can be found
+	 */
+    @Test
+    public void testStandaloneAnalysis() {
         var allSequences = analysis.findAllSequences();
         allSequences.stream()
             .map(it -> it.toString())
             .forEach(System.out::println);
 
         assertFalse(allSequences.isEmpty());
     }
-
+    
+    /**
+     * Tests the expected amount of ActionSequences. 
+     * <p> Fails if the analysis does not find two sequences
+     */
     @Test
-    public void testPCMActionSequenceFinder() {
-        final var usageModelPath = Paths.get("models", "BranchingOnlineShop", "default.usagemodel").toString();
-        final var allocationPath = Paths.get("models", "BranchingOnlineShop", "default.allocation").toString();
-
-        final DataFlowConfidentialityAnalysis analysis = new StandalonePCMDataFlowConfidentialtyAnalysis(
-                TEST_MODEL_PROJECT_NAME, Activator.class, usageModelPath, allocationPath);
+    public void testPCMActionSequenceFinderCount() {
+    	var allSequences = analysis.findAllSequences();
+    	assertEquals(2, allSequences.size(), String.format("Expected two dataflow sequences, but found %s sequences", allSequences.size()));
+    }
 
-        analysis.initalizeAnalysis();
+    /**
+     * Tests the expected sequence of elements
+     * <p> Fails if the analysis does not find the correct Classes for the first sequence
+     */
+    @Test
+    public void testPCMActionSequenceFinderTypes() {
         var allSequences = analysis.findAllSequences();
 
-        assertTrue(allSequences.size() == 2);
+        assertEquals(allSequences.size(), 2);
 
-        assertSequenceElements(allSequences.get(0), CallingUserActionSequenceElement.class,
-                CallingSEFFActionSequenceElement.class, SEFFActionSequenceElement.class,
-                CallingSEFFActionSequenceElement.class, SEFFActionSequenceElement.class,
-                CallingUserActionSequenceElement.class, CallingUserActionSequenceElement.class,
-                CallingSEFFActionSequenceElement.class, CallingSEFFActionSequenceElement.class,
-                CallingUserActionSequenceElement.class);
+        assertSequenceElements(allSequences.get(0), 
+        		CallingUserActionSequenceElement.class, // UserAction-Call: ViewEntryLevelSystemCall
```

Comment:
As comments have no semantics, these comments could become outdated quite easily. I would suggest either
a) Remove the comments (although I appreciate the effort :))
b) Create separate test cases that check for validity, e.g., of the names in named entities. This could build on the functions for checking the element content you already added.

Reply:
Nevertheless, I would not focus on such detailed test cases for now. So maybe go with b) only if you have spare time besides finishing the analysis implementation :)

Reply:
I would suggest removing them for now, but creating separate test cases in the future. For now i would like to get this PR merged, so i can implement the tests for the label propagation in the other pull request